src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,78 @@ def query_hook(self, event, start, end, environment_id):
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        project_id = event.project_id
+        cache_key = f"r.c.spc:{project_id}-{environment_id}".replace(" ", "")
+        session_count_last_hour = cache.get(cache_key)
+        if session_count_last_hour is None:
+            filters = {"project_id": [project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+            result_totals = raw_query(
+                selected_columns=["sessions"],
+                rollup=60,
+                dataset=Dataset.Sessions,
+                start=end - timedelta(minutes=60),
+                end=end,
+                filter_keys=filters,
+                groupby=["bucketed_started"],
+                referrer="rules.conditions.event_frequency.EventFrequencyPercentCondition",
+            )
+            if result_totals["data"]:
+                session_count_last_hour = 0
+                for bucket in result_totals["data"]:
+                    session_count_last_hour += bucket["sessions"]
+            else:
+                session_count_last_hour = False
+            cache.set(cache_key, session_count_last_hour, 600)
+
+        if session_count_last_hour:
+            interval_in_minutes = (
+                end - start
+            ).total_seconds() / 60  # percent_intervals[self.get_option("interval")][1].total_seconds() / 60
+            avg_sessions_in_interval = session_count_last_hour / (60 / interval_in_minutes)
+            issue_count = self.tsdb.get_sums(
+                model=self.tsdb.models.group,
+                keys=[event.group_id],
+                start=start,
+                end=end,
+                environment_id=environment_id,
+                use_cache=True,
+            )[event.group_id]
+            return 100 * round(issue_count / avg_sessions_in_interval, 4)
```

Comment:
I think @adhiraj mentioned this in slack, but it probably does make sense to make sure `issue_count < avg_sessions_in_interval`. Doesn't have to be in this pr